runtime/autoload/man.vim

```diff
@@ -91,7 +91,7 @@ function! s:read_page(sect, name) abort
   let b:manwidth = s:manwidth()
   silent execute 'read!env MANWIDTH='.b:manwidth s:man_cmd s:man_args(a:sect, a:name)
   " remove all the backspaced text
-  silent keeppatterns keepjumps %substitute,.\b,,ge
+  silent keeppatterns keepjumps execute '%substitute,.\b,,e'.(&gdefault?'':'g')
```

Comment:
Not sure but you might want to put `keeppatterns keepjumps` immediately before the substitute now.